src/software/simulated_tests/simulated_tests_test.cpp

```diff
@@ -0,0 +1,344 @@
+#include <gtest/gtest-spi.h>
+#include <gtest/gtest.h>
+
+#include "software/simulated_tests/mock_ai_wrapper.h"
+#include "software/simulated_tests/simulated_test_fixture.h"
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/test_util/test_util.h"
+#include "software/util/logger/init.h"
+#include "software/util/time/duration.h"
+#include "software/util/time/timestamp.h"
+#include "software/world/world.h"
+
+/**
+ * This class replaces the AIWrapper from the SimulatedTest with a MockAIWrapper,
+ * so that any AI logic does not affect these tests.
+ *
+ * These tests are designed to validate that all the components of the simulated testing
+ * system work together as expected. This includes the SimulatorBackend simulating and
+ * publishing the world correctly, the WorldStateValidator performing checks as expected,
+ * and the flow of control going through the AI and then back to the SimulatedBackend. We
+ * use the MockAIWrapper so that we are not testing any behaviour relating to the AI,
+ * just the testing infrastructure.
+ */
+class MockSimulatedTest : public SimulatedTest
+{
+   protected:
+    void SetUp() override
+    {
+        Util::Logger::LoggerSingleton::initializeLogger();
+        backend = std::make_shared<SimulatorBackend>(
+            Duration::fromMilliseconds(5), Duration::fromSeconds(1.0 / 30.0),
+            SimulatorBackend::SimulationSpeed::FAST_SIMULATION);
+        world_state_validator = std::make_shared<WorldStateValidator>();
+        mock_ai_wrapper       = std::make_shared<MockAIWrapper>();
+
+        backend->Subject<World>::registerObserver(world_state_validator);
+        world_state_validator->Subject<World>::registerObserver(mock_ai_wrapper);
+        mock_ai_wrapper->Subject<ConstPrimitiveVectorPtr>::registerObserver(backend);
+    }
+
+    std::shared_ptr<MockAIWrapper> mock_ai_wrapper;
+};
+
+TEST_F(MockSimulatedTest, test_single_validation_function_passes_before_timeout)
+{
+    World world         = ::Test::TestUtil::createBlankTestingWorld();
+    world.mutableBall() = Ball(Point(0, 0), Vector(4, 1.5), Timestamp::fromSeconds(0));
+
+    std::vector<ValidationFunction> validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            while (world_ptr->ball().position().x() < 3)
+            {
+                yield();
+            }
+        }};
+
+    std::vector<ValidationFunction> continous_validation_functions = {};
+
+    backend->startSimulation(world);
+    bool test_passed = world_state_validator->waitForValidationToPass(
+        validation_functions, continous_validation_functions, Duration::fromSeconds(1));
+    EXPECT_TRUE(test_passed);
+}
```

Comment:
Due to complexity? Are the names not descriptive enough? Is the code not self-explanatory enough? Not necessarily against this just want to clarify in case all simulated tests will need this in the future.

Reply:
Ah, so the reason that I think a comment is helpful here is that we're testing this is a somewhat roundabout way. If I'm looking at this for the first time, I might be confused why we're doing stuff with the ball for this super specific simulator validation test. How does this validation function test what we want it to test? Without doing some math and noting the initial ball velocity, it's really not obvious. 

Might be helpful if we could change the validation function here to just be a function of a timestamp? Ie "if past this time then done"? Just trying to avoid people having to think about the physics of the ball in a simulator test.

Similarly for the below tests.

Reply:
updated